lib/src/ui/common/mixin/error_mixin.dart

```diff
@@ -0,0 +1,11 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+
+mixin ErrorMessage {
+  Widget buildErrorMessage(FirebaseAuthException error) => error != null
```

Comment:
jako, że jest w pakiecie common to powinien być bardziej generyczny tj. przyjmować `Exception`, chyba, że ma być typowo pod FirebaseAuthException, wówczas chyba powinien być gdzieś w pakiecie `authentication` i ewentualnie mógłby nazywać się `AuthenticationErrorMessage` i odpowiednio `buildAuthenticationErrorMessage`

Reply:
to błąd, który będzie raczej używany tylko przy autentykacji, więc wybiorę tą drugą opcję